providers/amazon/tests/unit/amazon/aws/executors/aws_lambda/docker/test_app.py

```diff
@@ -0,0 +1,440 @@
+        message_body = json.loads(call_args[1]["MessageBody"])
+        assert message_body[TASK_KEY_KEY] == task_key
+        assert message_body[RETURN_CODE_KEY] == 0
+
+    def test_run_and_report_string_command(self, mock_sqs_client):
+        """Test run_and_report with string command (shell=True)."""
+        # Setup
+        command = "airflow version"  # String command
+        task_key = "test-task-key"
+
+        with patch("airflow.providers.amazon.aws.executors.aws_lambda.docker.app.subprocess.run") as mock_run:
+            mock_result = MagicMock()
+            mock_result.returncode = 0
+            mock_result.stdout = b"Airflow version output"
+            mock_run.return_value = mock_result
+
+            # Execute
+            run_and_report(command, task_key)
+
+            # Assert - shell should be True for string commands
+            mock_run.assert_called_once_with(
+                command,
+                check=False,
+                shell=True,  # command is a string, so shell=True
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
+
+    def test_run_and_report_sqs_exception(self, mock_sqs_client, mock_subprocess_run, caplog):
+        """Test run_and_report when SQS send_message raises an exception."""
+        # Setup
+        command = ["airflow", "version"]
+        task_key = "test-task-key"
+
+        mock_sqs = mock_sqs_client.return_value
+        mock_sqs.send_message.side_effect = Exception("SQS failed")
+
+        # Execute
+        with caplog.at_level("ERROR"):
+            run_and_report(command, task_key)
+
+        # Assert
+        mock_subprocess_run.assert_called_once()
+        mock_sqs_client.assert_called_once()
+        mock_sqs.send_message.assert_called_once()
+        # Check that error was logged
+        assert "Failed to send message to SQS" in caplog.text
+
+    def test_get_sqs_client(self):
+        """Test get_sqs_client function."""
+        # Setup
+        with patch("airflow.providers.amazon.aws.executors.aws_lambda.docker.app.boto3.client") as mock_boto:
+            mock_client = MagicMock()
+            mock_boto.return_value = mock_client
+
+            # Execute
+            result = get_sqs_client()
+
+            # Assert
+            mock_boto.assert_called_once_with("sqs")
+            assert result == mock_client
+
+    def test_get_queue_url_from_env_var(self):
+        """Test get_queue_url with AIRFLOW__AWS_LAMBDA_EXECUTOR__QUEUE_URL environment variable."""
+        # Setup
+        test_url = "https://sqs.us-east-1.amazonaws.com/123456789012/test-queue"
+        with patch.dict(os.environ, {"AIRFLOW__AWS_LAMBDA_EXECUTOR__QUEUE_URL": test_url}):
+            # Execute
+            result = get_queue_url()
+
+            # Assert
+            assert result == test_url
+
+    def test_get_queue_url_from_legacy_env_var(self):

```

Comment:
You can at least `@pytest.mark.parametrize(...)` this test with the one above. And likely also the one below if you add a condition that checks if there url string is empty and then expect an exception (but fair if you want to stay somewhat DAMP and keep that one separate).